crypto/bindings/conversion.ts

```diff
@@ -98,3 +158,120 @@ function fieldsFromRustFlat(fieldBytes: Uint8Array): MlArray<Field> {
   }
   return [0, ...fields];
 }
+
+// affine
+
+type WasmAffine = wasmNamespace.WasmGVesta | wasmNamespace.WasmGPallas;
+type MakeAffine<A extends WasmAffine> = () => A & { ptr: number };
+
+function affineFromRust(pt: WasmAffine): OrInfinity {
+  if (pt.infinity) {
+    pt.free();
+    return 0;
+  } else {
+    let x = fieldFromRust(pt.x);
+    let y = fieldFromRust(pt.y);
+    pt.free();
+    return [0, [0, x, y]];
+  }
+}
+
+function affineToRust<A extends WasmAffine>(
+  pt: OrInfinity,
+  klass: MakeAffine<A>
+) {
+  var res = klass();
+  if (pt === 0) {
```

Comment:
If you change `OrInfinity` as discussed [here](https://github.com/o1-labs/o1js-bindings/pull/140/files#r1329991174), it will still work as by luck `Infinity` is the first element of the ADT, but I would use a helper instead.

Reply:
it doesn't work by luck - it works because this is all strictly typed and the type of `pt` already correctly reflected the runtime value. TS wouldn't have let us access `pt[1]` below if we didn't handle the `pt === 0` case here.

I improved readability of this code by adding a constant `Infinity = 0`. Other than that, I think handling the different possibilities of an enum with if statements, like this function does, is the correct TS analogue of matching an enum in OCaml. It's type-safe and it is idiomatic JS code. I don't want to introduce a separate helper function for every enum just to express a match statement.